test/e2e/reconciler/ingress/controller_test.go

```diff
@@ -93,9 +94,14 @@ func TestIngressController(t *testing.T) {
 						return false
 					}
 					if len(got.Items) != 1 {
+						klog.Errorf("expected 1 ingress, got %d", len(got.Items))
+						return false
+					}
+					diff := cmp.Diff(got.Items[0].Spec, rootIngress.Spec)
+					if diff != "" {
+						klog.Errorf("Expected no diff, got %s", diff)
 						return false
 					}
-					framework.RequireNoDiff(t, got.Items[0].Spec, rootIngress.Spec)
```

Comment:
Maybe this was the problem? If the data wasn't correct the first time it was seen, this check would fail the test immediately when what we actually want is to retry?

Reply:
I think`require` statements in polling loops may be an anti-pattern. If you need to fail out of a loop, it's probably better to use one of the `wait.Poll*` methods that support returning an error.